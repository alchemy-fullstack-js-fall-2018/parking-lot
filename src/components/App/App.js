import React, { Fragment } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from 'react-router-dom';
import ROUTES from '../../routes';
import Header from '../Header/Header';

export default function App() {
  return (
    <Router>
      <Fragment>
        <Header />
        <Switch>
          <Route component={ROUTES.SEARCH_CARS.Component} path={ROUTES.SEARCH_CARS.path} />
          <Route component={ROUTES.REGISTER_CAR.Component} path={ROUTES.REGISTER_CAR.path} />
          <Route component={ROUTES.CAR.Component} path={ROUTES.CAR.path} />
          <Route component={ROUTES.ALL_CARS.Component} path={ROUTES.ALL_CARS.path} />
          <Route component={ROUTES.SEARCH_LOT_CARS.Component} path={ROUTES.SEARCH_LOT_CARS.path} />
          <Route component={ROUTES.LOT_CARS.Component} path={ROUTES.LOT_CARS.path} />
          <Route component={ROUTES.HOME.Component} path={ROUTES.HOME.path} />
        </Switch>
      </Fragment>
    </Router>
  );
}
