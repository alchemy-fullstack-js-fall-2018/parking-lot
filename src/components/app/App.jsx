import React, { Fragment } from 'react';
import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';
import { ROUTES } from '../../routes';
import { Helmet } from 'react-helmet';
import { createGlobalStyle } from 'styled-components';
import 'normalize.css';

import Header from './Header.jsx';
import Footer from './Footer.jsx';

const GlobalStyle = createGlobalStyle`
  #root {
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: space-between;
    text-align: center;
    margin: 0;
  }
`;

const App = () => {

  return (
    <Router>
      <Fragment>
        <Helmet>
          <title>Cars</title>
        </Helmet>
        <GlobalStyle/>
        <Header />
        <Switch>
          <Route exact component={ROUTES.ABOUT.Component} path={ROUTES.ABOUT.path} />
          <Route exact component={ROUTES.HOME.Component} path={ROUTES.HOME.path} />
          <Route exact component={ROUTES.CARS.Component} path={ROUTES.CARS.path} />
          <Route exact component={ROUTES.LOT.Component} path={ROUTES.LOT.path} />
          <Route exact component={ROUTES.CAR_FORM.Component} path={ROUTES.CAR_FORM.path} />
          <Route exact component={ROUTES.CAR_SEARCH_ALL.Component} path={ROUTES.CAR_SEARCH_ALL.path} />
          <Route exact component={ROUTES.CAR_SEARCH_LOT.Component} path={ROUTES.CAR_SEARCH_LOT.path} />
          <Route component={ROUTES.CAR_DETAIL.Component} path={ROUTES.CAR_DETAIL.path} />
          <Redirect to={ROUTES.HOME.path} />
        </Switch>
        <Footer />
      </Fragment>
    </Router>
  );
};

export default App;
