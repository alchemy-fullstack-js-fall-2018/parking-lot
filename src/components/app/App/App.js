import React, { Fragment } from 'react';
import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';
import { ROUTES } from '../../../routes';

import Header from '../Header';
import Footer from '../Footer';

import 'normalize.css';
import styles from './App.css';

const App = () => {

  return (
    <Router>
      <Fragment>
        <Header />
        <Switch>
          <Route component={ROUTES.ABOUT.Component} path={ROUTES.ABOUT.path} />
          <Route component={ROUTES.HOME.Component} path={ROUTES.HOME.path} />

          <Redirect to={ROUTES.HOME.path} />
        </Switch>
        <Footer />
      </Fragment>
    </Router>
  );
};

export default App;
