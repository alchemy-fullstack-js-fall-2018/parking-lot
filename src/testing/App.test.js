import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/app/App.jsx';


describe('<App />', () => {
  test('renders', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
