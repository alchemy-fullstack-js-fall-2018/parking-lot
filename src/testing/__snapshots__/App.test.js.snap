// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<BrowserRouter>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Cars
    </title>
  </HelmetWrapper>
  <GlobalStyleComponent
    suppressMultiMountWarning={false}
  />
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/about/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/home/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/cars/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/lot/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/cars/new"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/cars/search"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/lot/search"
    />
    <Route
      component={[Function]}
      path="/cars/:id"
    />
    <Redirect
      push={false}
      to="/home/"
    />
  </Switch>
  <Footer />
</BrowserRouter>
`;
